server/src/Metrics.hs

```diff
@@ -4,13 +4,16 @@ import           Prometheus
 import qualified Network.HTTP.Types as Http
 import qualified Data.ByteString.Char8 as BS8
 
-type MethodLabel = String
-type StatusLabel = String
+type HttpMethodLabel = String
+type HttpStatusLabel = String
 
-type RequestCounter = Vector (MethodLabel, StatusLabel) Counter
+type HttpRequestCounter = Vector (HttpMethodLabel, HttpStatusLabel) Counter
+
+countHttpRequest :: Http.Method -> Http.Status -> Metric HttpRequestCounter -> IO ()
+countHttpRequest method status = withLabel (BS8.unpack method, show $ Http.statusCode status) incCounter
```

Comment:
Sorry man, I meant to inline the `where` instead of having it in a new line:

``` haskell
foo = bar
  where bar = this
```

Instead of

``` haskell
foo = bar
  where
     bar = this
```

Reply:
Ok, I see that you created this from the where clause of the other function. I think it's alright, but move this function underneath the `notifyRequest` function so improve code proximity and reading travel effort.

Reply:
In terms of code proximity, I'd keep it this way, `countHttpRequest` is more general and works on `HttpRequestCounter` metrics, while all the other functions below are working on the aggregate `IcepeakMetrics`.

Reply:
Aright, fair enough.